packages/indexer-common/src/indexer-management/monitor.ts

```diff
@@ -549,11 +555,113 @@ export class NetworkMonitor {
     return deployments
   }
 
+  async latestValidEpoch(network?: string): Promise<NetworkEpochBlock> {
+    network = network ?? this.blockOracleSubgraph.network
+    this.logger.trace(`latest valid epoch`, {
+      settlementNetwork: this.blockOracleSubgraph.network,
+      network,
+      networkAlias: NetworkAliases[network],
+    })
+    try {
+      const result = await this.blockOracleSubgraph.query(
+        gql`
+          query globalStates($network: String!) {
+            globalStates {
+              latestValidEpoch {
+                epochNumber
+                blockNumbers(where: { network: $network }) {
+                  network {
+                    id
+                  }
+                  blockNumber
+                  previousBlockNumber {
+                    blockNumber
+                  }
+                }
+              }
+            }
+          }
+        `,
+        {
+          network: NetworkAliases[network],
+        },
+      )
+
+      if (result.error) {
+        throw result.error
+      }
+
+      if (!result.data.globalStates) {
+        throw indexerError(
+          IndexerErrorCode.IE069,
+          `Failed to query latest valid epoch number for the global state`,
+        )
+      }
+
+      const epochBlock = await buildEpochBlock(
+        this.indexingStatusResolver,
+        result.data.globalStates[0].latestValidEpoch.epochNumber,
+        result.data.globalStates[0].latestValidEpoch.blockNumbers[0],
+      )
+
+      return epochBlock
+    } catch (err) {
+      this.logger.error(
+        `Failed to query latest global epoch number from epoch block oracle`,
+        {
+          err,
+        },
+      )
+      throw err
+    }
+  }
+
+  async fetchPOIBlockPointer(allocation: Allocation): Promise<BlockPointer> {
+    try {
+      const networkID = (
+        await this.indexingStatusResolver.indexingStatus([
+          allocation.subgraphDeployment.id,
+        ])
+      )[0].chains[0].network
```

Comment:
Here also, would love to use something more clear than [0] indexing if it makes sense.

Reply:
There isn't a single element version of the query from the schema as far as I could tell, so I added a query to `indexingStatusForCurrentVersion` with subgraph name